### PR TITLE
fixing a few issues (see Makefile for details)

### PR DESCRIPTION
**Pull request by intheswim**
- fixing issue with not showing inside xscreensaver window (unless one opens Preview)
- making it respond to screen size changes (as a standalone app or inside xscreensaver)
- handling "window close" event and doing cleanup part.

### CODE REVIEW

**Comment by RobertZenz** [comment]
Hot damn that's a long time ago.

**Comment by RobertZenz** [comment]
From quickly looking at it, without verifying the logic, but does look okay on the first look. Thanks.

---

I would move the Makefile one level higher, so that it resides at the project root.

---

You could add a formatting target, like this;

```Makefile
format:
	astyle \
		--align-pointer=type \
		--align-reference=type \
		--fill-empty-lines \
		--indent=force-tab=4 \
		--indent-switches \
		--min-conditional-indent=2 \
		--suffix=none \
		src/*.c
```

That's the style I've adopted for pretty much all later projects (yes, you can take my Java style out of me cold, dead hands). That would allow to make sure that the files are always formatted the same, no matter who writes them.

File: lavanet/src/Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+CC=gcc
+
+# original version was not showing in xscreensaver little window;
+# it also did not respond to resize window events (for example when run from terminal).
+
+CFLAGS = -Wall -O2
+CLIBS = -lX11 -lm
+
+all : lavanet.c vroot.h
+		$(CC) $(CFLAGS) -o lava lavanet.c $(CLIBS)
```
Comment (RobertZenz) [inline lavanet/src/Makefile]: I would call the executable the same as the project, `lavanet`.

That's mainly to cut down on confusion, if everything always has the same name you never need to start mapping names in your head.

Comment (intheswim) [reply lavanet/src/Makefile]: No problem with that, I just like shorter names :)

**Comment by intheswim** [comment]
There's an interesting visual effect now when you resize (downsize) the window, which I have not really expected, but it's worth checking. I am on Ubuntu 18.04.

**Comment by RobertZenz** [comment]
The effect which is going on is simply how the points locations are being calculated. If you increase the window size, the points will slowly start wandering around, with the exception of those that leave through the edges, they will warped to the other side.

Otherwise looks like it's working, thank you.

**Pull request merged**